core/kernel/ldelf_loader.c

```diff
@@ -452,5 +496,10 @@ TEE_Result ldelf_dlsym(struct user_mode_ctx *uctx, TEE_UUID *uuid,
 			*val = arg->dlsym.val;
 	}
 
+out_free_dl_entry_arg:
```

Comment:
note here you can simplify by dropping branch label `out_free_dl_entry_arg` and only use `out` as branch label in error cases above. This is permitted as `free(NULL)` is valid:
```c
out:
	free(dl_entry_arg);
	free(uuid_kbuf);

	return res;
```